### DISCUSSION TITLE
SQLite with private mode

### DISCUSSION THREAD

**Discussion by steida** [rhashimoto/wa-sqlite#161] [General]
OPFS isn't enabled in Firefox (and probably Safari) private mode, but it should be possible to use in-memory SQLite anyway. Did someone explore that issue? Thank you

**Comment by rhashimoto** [comment]
You can just try the [demo](https://rhashimoto.github.io/wa-sqlite/demo/). Seems to work for me.

**Reply by steida** [reply]
AccessHandlePool isn't working in private mode in Firefox and Safari macOS, at least for me.

**Reply by rhashimoto** [reply]
I thought you were asking about in-memory SQLite. If OPFS isn't enabled then AccessHandlePoolVFS isn't going to work.

**Reply by steida** [reply]
@rhashimoto Btw, did you test AccessHandlePoolVFS with bfcache? Or maybe just going from one URL to another URL of the same domain? I'm using opfs-shapool (as you probably know), and URL navigation breaks it in SAFARI :-/ I'm going to try your lib under the same conditions. What is interesting and also very unfortunate is that Safari is buggy only when the dev inspector IS NOT shown. When I open the dev inspector, everything works, which makes it super hard to debug.

**Reply by rhashimoto** [reply]
I didn't even know about bfcache, so no, I haven't tested it. I have on occasion used multiple URLs with the same origin but I don't specifically remember if I tried AccessHandlePoolVFS (or any other OPFS VFS) on Safari.
